staging/src/k8s.io/apiserver/plugin/pkg/authenticator/token/oidc/oidc.go

```diff
@@ -104,6 +113,92 @@ type Options struct {
 
 	// now is used for testing. It defaults to time.Now.
 	now func() time.Time
+
+	// IssuersPerClaim supports distributed claims.
+	//
+	// See for details:
+	// http://openid.net/specs/openid-connect-core-1_0.html#AggregatedDistributedClaims
+	//
+	// Lists, for each claim, the list of issuer URLs that are
+	// trusted to provide the respective distributed claim.  Only distributed
+	// claims with issuers as specified below will be resolved.  This is done
+	// to reduce the chance that an authentication request goes to an address
+	// that we don't trust to issue distributed claims.
+	//
+	// Multiple issuers are supported so that it would be possible to rotate
+	// the issuers over time (with sufficient advance notice) without
+	// disrupting cluster operations, while it is expected that more than one
+	// issuer per claim is present only if the distributed claims source change
+	// is imminent.
+	//
+	// Example:
+	// "groups" -> []string{"http://example1.com/foo","http://example2.com/bar"}
+	IssuersPerClaim map[string][]string
+}
+
+// initVerifier creates a new ID token verifier for the given configuration and issuer URL.  On success, calls setVerifier with the
+// resulting verifier.  Returns true in case of success, or non-nil error in case of an irrecoverable error.
+func initVerifier(ctx context.Context, config *oidc.Config, iss string) (*oidc.IDTokenVerifier, error) {
+	provider, err := oidc.NewProvider(ctx, iss)
+	if err != nil {
+		return nil, fmt.Errorf("init verifier failed: %v", err)
+	}
+	return provider.Verifier(config), nil
+}
+
+// asyncIDTokenVerifier is an ID token verifier that allows async initialization
+// of the issuer check.  Must be passed by reference as it wraps sync.Mutex.
+type asyncIDTokenVerifier struct {
+	m sync.Mutex
+
+	// v is the ID token verifier initialized asynchronously.  It remains nil
+	// up until it is eventually initialized.
+	// Guarded by m
+	v *oidc.IDTokenVerifier
+}
+
+// newAsyncIDTokenVerifier creates a new asynchronous token verifier.  The
+// verifier is available immediately, but may remain uninitialized for some time
+// after creation.
+// TODO: newAuthenticator could use this as well.
+func newAsyncIDTokenVerifier(ctx context.Context, c *oidc.Config, iss string) *asyncIDTokenVerifier {
+	t := &asyncIDTokenVerifier{}
+
+	// Polls indefinitely in an attemt to initialize the distributed claims
+	// verifier, or until context canceled.
+	sync := make(chan struct{})
+	initFn := func() (done bool, err error) {
```

Comment:
err is always nil. Should this be `func() bool` ?

Reply:
It has to conform to `wait.PollUntil` API.